implementation/src/main/java/io/smallrye/config/ConfigSourceInterceptor.java

```diff
@@ -32,4 +34,29 @@
      * @return a ConfigValue with information about the key name, value, config source origin and ordinal.
      */
     ConfigValue getValue(ConfigSourceInterceptorContext context, String name);
+
+    default Iterator<String> getNames(ConfigSourceInterceptorContext context) {
```

Comment:
Maybe these should be `iterateNames` for consistency with the context (see other comment)?